hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/validate/HazelcastSqlOperatorTable.java

```diff
@@ -402,6 +408,40 @@ private SqlCall rewriteCall(SqlCall call) {
             return call;
         }
 
+        @SuppressWarnings("CheckStyle")
+        private SqlCall expandView(SqlCall call) {
+            if (call instanceof SqlSelect) {
+                SqlSelect selectCall = (SqlSelect) call;
+                ViewResolver viewResolver = validator.getViewResolver();
+
+                if (selectCall.getFrom() == null) {
+                    return call;
+                }
+
+                SqlNode from = selectCall.getFrom();
+                if (from instanceof SqlIdentifier) {
+                    SqlIdentifier fromClause = (SqlIdentifier) from;
+                    if (!ValidationUtil.isCatalogObjectNameValid(fromClause)) {
+                        // We are not throwing any exceptions here, delegating it to validation stage.
+                        return call;
+                    }
+                    String id = fromClause.names.get(fromClause.names.size() - 1);
+                    View resolvedView = viewResolver.resolve(id);
+                    if (resolvedView == null) {
+                        return call;
+                    } else {
+                        QueryParser parser = new QueryParser(validator);
+                        // Note: query was parsed & validated previously.
+                        // We assume, that no exception happens here during parsing.
```

Comment:
Error might happen inside. We also validate inside, the target object can be dropped in the meantime. We need to wrap the exception and make sure the error message makes sense.